Core/Core/View/Base/AppReview/AppReviewViewModel.swift

```diff
@@ -0,0 +1,143 @@
+    
+    enum ReviewState {
+        case vote
+        case feedback
+        case thanksForVote
+        case thanksForFeedback
+        
+        var title: String {
+            switch self {
+            case .vote:
+                CoreLocalization.Review.voteTitle
+            case .feedback:
+                CoreLocalization.Review.feedbackTitle
+            case .thanksForVote, .thanksForFeedback:
+                CoreLocalization.Review.thanksForVoteTitle
+            }
+        }
+        
+        var description: String {
+            switch self {
+            case .vote:
+                CoreLocalization.Review.voteDescription
+            case .feedback:
+                CoreLocalization.Review.feedbackDescription
+            case .thanksForVote:
+                CoreLocalization.Review.thanksForVoteDescription
+            case .thanksForFeedback:
+                CoreLocalization.Review.thanksForFeedbackDescription
+            }
+        }
+    }
+    
+    @Published var state: ReviewState = .vote
+    @Published var rating: Int = 0
+    @Published var showReview: Bool = false
+    @Published var showSelectMailClientView: Bool = false
+    @Published var feedback: String = ""
+    @Published var clients: [ThirdPartyMailClient] = []
+    let allClients = ThirdPartyMailClient.clients
+    
+    private let config: Config
+    private var storage: CoreStorage
+    
+    public init(config: Config, storage: CoreStorage) {
+        self.config = config
+        self.storage = storage
+    }
+    
+    public func shouldShowRatingView() -> Bool {
+        guard let currentVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String else {
+            return false
+        }
+        guard let lastShownVersion = storage.reviewLastShownVersion else {
+            storage.reviewLastShownVersion = currentVersion
+            return true
+        }
+        return isNewerVersion(currentVersion: currentVersion, lastVersion: lastShownVersion)
+    }
+    
+    func reviewAction() {
+        withAnimation(Animation.easeIn(duration: 0.2)) {
+            if rating <= 3 {
+                state = .feedback
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
+                    withAnimation(Animation.easeIn(duration: 0.1)) {
+                        self.showReview = true
+                    }
+                }
+            } else {
+                state = .thanksForVote
+            }
+        }
+    }
+    
+    func writeFeedbackToMail() {
+        self.clients = allClients.filter({ ThirdPartyMailer.isMailClientAvailable($0) })
+        if !clients.isEmpty {
+            withAnimation(Animation.bouncy(duration: 0.2)) {
+                showSelectMailClientView = true
+            }
+        } else {
+            openMailClient(ThirdPartyMailClient.systemDefault)
+        }
+    }
+    
+    func requestReview() {
+        SKStoreReviewController.requestReview()
```

Comment:
Hello @IvanStepanok 
As I know SKStoreReviewController can be showed three times per year only
https://developer.apple.com/documentation/storekit/requesting_app_store_reviews
All other times 'Rate Us' button won't work, right?
Thank you